### ISSUE TITLE
Too aggressive syncing when window focus change

### ISSUE THREAD

**Issue by sandy081** [microsoft/vscode#91556]
From @joaomoreno 

@sandy081 @Tyriar Since auto sync runs every time the window gets focused, this can happen too often. I could get it to run every 2 seconds by focusing in/out of the window. Not sure whether you want to put a rate limiter there, since it will generate lots of traffic once every user uses it.

**Comment by sandy081** [comment]
Agreed. May be we should not be too aggressive on focus change. We can restrict this by at most one sync request per minute on focus change.

**Comment by Tyriar** [comment]
Sounds fine, I guess we can extend this to opening the settings.json as well, we might want a "Sync now" item in the "Sync is on" quick pick though to allow users to force a sync? This is probably needed if we're going to be rate limiting it.

**Comment by Tyriar** [comment]
We might want to fix this for insiders next week before Wednesday.

**Comment by sandy081** [comment]
I can look into this tomorrow and see if it is trivial.

BTW my understanding is that rolling out to insiders has flexibility of daily fixes and updates and this is not too critical to fix unless you think this will make the server too busy.

**Comment by sandy081** [comment]
Is not this seen because of testing. I just checked my sync log and in the last one hour, sync is triggered 30 times which means on an avg once in 2 mins. So are we over seeing this?

So, it would be good to get the reports first before speculating. Added telemetry from the client side to see how often auto sync is triggered from different source.

**Comment by Tyriar** [comment]
I've certainly had times where I've fired off a lot of requests, depends what that looks like at scale. You're probably right in that we can look at it after we go out without additional problems.

**Comment by sandy081** [comment]
Non sync triggers (window focus change, new window, opening settings, keybindings etc) can sync only if sync did not happen in last one minute. In other words they can sync max once per minute.
